packages/mobile/src/account/profileInfo.ts

```diff
@@ -0,0 +1,317 @@
+    const dataURL = getDataURL(mimeType, data)
+    const pictureAccessor = new PrivatePictureAccessor(offchainWrapper)
+    writeError = yield call([pictureAccessor, 'write'], dataURL, [])
+    Logger.info(TAG + 'uploadNameAndPicture', 'uploaded profile picture')
+  }
+
+  if (writeError) {
+    Logger.error(TAG + '@uploadNameAndPicture', writeError)
+    throw Error('Unable to write data')
+  }
+}
+
+// this function gives permission to the recipient to view the user's profile info
+export function* uploadSymmetricKeys(recipientAddresses: string[]) {
+  // TODO: check if key for user already exists, skip if yes
+  const account = yield select(currentAccountSelector)
+  const contractKit = yield call(getContractKit)
+  yield call(unlockDEK)
+  const offchainWrapper = new UploadServiceDataWrapper(contractKit, account)
+
+  const nameAccessor = new PrivateNameAccessor(offchainWrapper)
+  let writeError = yield call([nameAccessor, 'allowAccess'], recipientAddresses)
+
+  const pictureUri = yield select(pictureSelector)
+  if (pictureUri) {
+    const pictureAccessor = new PrivatePictureAccessor(offchainWrapper)
+    writeError = yield call([pictureAccessor, 'allowAccess'], recipientAddresses)
+  }
+
+  Logger.info(TAG + '@uploadSymmetricKeys', 'uploaded symmetric keys for ' + recipientAddresses)
+
+  if (writeError) {
+    Logger.error(TAG + '@uploadSymmetricKeys', writeError)
+    throw Error('Unable to write keys')
+  }
+}
+
+export function* getProfileInfo(address: string) {
+  // TODO: check if we already have profile info of address
+  const account = yield select(currentAccountSelector)
+  const contractKit = yield call(getContractKit)
+  yield call(unlockDEK)
+  const offchainWrapper = new UploadServiceDataWrapper(contractKit, account)
+
+  const nameAccessor = new PrivateNameAccessor(offchainWrapper)
+  try {
+    const name = yield call([nameAccessor, 'read'], address)
+
+    const pictureAccessor = new PrivatePictureAccessor(offchainWrapper)
+    let picturePath
+    try {
+      const pictureObject = yield call([pictureAccessor, 'read'], address)
+      const pictureURL = pictureObject.toString()
+      picturePath = yield call(saveRecipientPicture, pictureURL, address)
+    } catch (error) {
+      Logger.warn("can't fetch picture for", address)
+    }
+    return { name: name.name, thumbnailPath: picturePath }
+  } catch (error) {
+    Logger.warn("can't fetch name for", address)
+  }
+}
+
+function* unlockDEK(addAccount = false) {
+  const privateDataKey: string | null = yield select(dataEncryptionKeySelector)
+  if (!privateDataKey) {
+    throw new Error('No data key in store. Should never happen.')
+  }
+  const dataKeyaddress = normalizeAddressWith0x(
+    privateKeyToAddress(ensureLeading0x(privateDataKey))
+  )
+  const wallet: UnlockableWallet = yield call(getWallet)
+  if (addAccount) {
+    try {
+      yield call([wallet, wallet.addAccount], privateDataKey, '')
+    } catch (e) {
```

Comment:
I'd imagine we want to log this error just in case something unexpected happens

Reply:
Yes we should log the real error.